include/circt/Dialect/Calyx/CalyxStructure.td

```diff
@@ -547,3 +618,42 @@ def GroupGoOp : CalyxGroupPort<"group_go", [
     }]>
   ];
 }
+
+def CycleOp : CalyxOp<"cycle", [
+  HasParent<"StaticGroupOp">,
+]> {
+  let summary = "Calyx Static Cycle Op";
+  let description = [{
+  }];
+  let arguments = (ins
+    I32Attr:$start,
+    OptionalAttr<I32Attr>:$end
```

Comment:
For repeat and cycle, are 32 bits enough? I guess probably...

Reply:
Yeah I made a guess here that 32 bits would be enough. I guess it's not a big deal to switch to 64 bits, but I'm not sure if it matters for any reasonable design.